edb/lang/graphql/translator.py

```diff
@@ -487,26 +482,51 @@ def get_path_prefix(self, end_trim=None):
         )
         return prefix
 
-    def _visit_arguments(self, args, *, get_path_prefix):
+    def visit_ListLiteral(self, node):
+        return qlast.Array(elements=self.visit(node.value))
+
+    def visit_InputObjectLiteral(self, node):
+        # this represents some expression to be used in filter
         result = []
-        for arg in args:
-            result.append(self.visit_Argument(
-                arg, get_path_prefix=get_path_prefix))
+        for field in node.value:
+            result.append(self.visit(field))
 
-        return result
+        return self._join_expressions(result)
 
-    def visit_Argument(self, node, *, get_path_prefix):
-        op = ast.ops.EQ
+    def visit_ObjectField(self, node):
         name_parts = node.name
 
+        # handle boolean ops
+        if name_parts == 'and':
+            return self._visit_list_of_inputs(node.value.value, ast.ops.AND)
+        elif name_parts == 'or':
+            return self._visit_list_of_inputs(node.value.value, ast.ops.OR)
+        elif name_parts == 'not':
+            return qlast.UnaryOp(op=ast.ops.NOT,
+                                 operand=self.visit(node.value))
+
+        # handle various scalar ops
+        op = gt.GQL_TO_OPS_MAP.get(name_parts)
+
+        if op:
+            value = self.visit(node.value)
+
+            if name_parts == 'contains':
+                value.value = f'%{value.value}%'
+            elif name_parts == 'startswith':
+                value.value = f'{value.value}%'
+            elif name_parts == 'endswith':
+                value.value = f'%{value.value}'
```

Comment:
These seem to be a special case of the "LIKE" operator. So I'd replace all of these with more generic "like" and "ilike" GraphQL filters.

Reply:
I've removed the "contains", "startswith" and "endswith" and replaced them with "like" and "ilike".